src/Spectre.libGaussianMixtureModelling/Matrix.h

```diff
@@ -31,29 +31,78 @@ struct Matrix
 {
     /// <summary>
     /// Constructor initializing the data based on given dimensions.
+    /// It sets all the values to zero upon creation.
     /// </summary>
     /// <param name="height">Height of the array.</param>
     /// <param name="width">Width of the array.</param>
     Matrix(unsigned height, unsigned width)
     {
+        unsigned length = height * width;
         data = new Type*[height];
         if (height > 0)
         {
-            data[0] = new Type[width * height]; // allocate data once
+            data[0] = new Type[length]; // allocate data once
             for (unsigned i = 1; i < height; ++i)
                 data[i] = data[0] + i * width; // assign pointers
         }
-        memset(data[0], 0, width * height * sizeof(Type));
+        memset(data[0], 0, length * sizeof(Type));
     }
 
+    /// <summary>
+    /// Construct a matrix from vector of vectors.
+    /// </summary>
+    /// <param name="other">Vector of vectors with data.</param>
+    Matrix(std::vector<std::vector<Type>>& other)
+    {
+        unsigned height = (unsigned)other.size();
+        unsigned width = height ? (unsigned)other[0].size() : 0;
+        unsigned length = height * width;
+        data = new Type*[height];
+        if (height > 0)
+        {
+            data[0] = new Type[length]; // allocate data once
+            for (unsigned i = 1; i < height; ++i)
+                data[i] = data[0] + i * width; // assign pointers
+        }
+        for (unsigned i = 0; i < height; i++)
+        {
+            for (unsigned j = 0; j < width; j++)
+            {
+                data[i][j] = other[i][j];
+            }
+        }
+    }
+
+    /// <summary>
+    /// Move constructor
+    /// </summary>
+    /// <param name="other">Old matrix object.</param>
+    Matrix(Matrix&& other)
+    {
+        data = other.data;
+        other.data = nullptr;
+    }
+
+    /// <summary>
+    /// Move constructor
+    /// </summary>
+    /// <param name="other">Old matrix object.</param>
+    Matrix(Matrix& other)
```

Comment:
This isn't move constructor and definitely should not be here. It's better to delete this.

Reply:
I have no idea how did this end up here, thanks for catching it.